src/features/events/components/EventContactCard.tsx

```diff
@@ -0,0 +1,129 @@
+  data: ZetkinEvent;
+  model: EventDataModel;
+  orgId: number;
+}
+
+const EventContactCard: FC<EventContactCardProps> = ({
+  data,
+  model,
+  orgId,
+}) => {
+  const messages = useMessages(messageIds);
+  const { showConfirmDialog } = useContext(ZUIConfirmDialogContext);
+
+  const handleSelectedPerson = (personId: number) => {
+    model.setContact(personId);
+  };
+
+  return (
+    <Box
+      mb={2}
+      sx={{
+        '& div': {
+          flexDirection: 'column',
+          flexWrap: 'wrap',
+        },
+        display: 'block',
+      }}
+    >
+      <ZUICard
+        header={
+          <Box>
+            {model.getData().data?.contact?.id ? (
+              <Box>
+                <FaceOutlinedIcon sx={{ margin: 1, verticalAlign: 'middle' }} />
+                {messages.eventContactCard.header()}
+              </Box>
+            ) : (
+              <Box>
+                <FaceRetouchingOffOutlinedIcon
+                  sx={{ margin: 1, verticalAlign: 'middle' }}
+                />
+                {messages.eventContactCard.noContact()}
+              </Box>
+            )}
+          </Box>
+        }
+        status={
+          <Box>
+            {data.contact?.id ? (
+              <>
+                <Box
+                  m={1}
+                  sx={{ display: 'inline-block', verticalAlign: 'middle' }}
+                >
+                  <ZUIAvatar orgId={orgId} personId={data.contact.id} />
+                </Box>
+                <Typography
+                  sx={{ display: 'inline-block', verticalAlign: 'middle' }}
+                >
+                  {data.contact.name}
+                </Typography>
+              </>
+            ) : (
+              <Box sx={{ display: 'grid' }}>
+                <Box
+                  m={1}
+                  mb={3}
+                  mt={3}
+                  sx={{
+                    '& input': {
+                      alignSelf: 'self-start',
+                      minWidth: '100px !important',
+                      width: '200px !important',
+                    },
+                  }}
+                >
+                  <ZUIPersonSelect
+                    onChange={(person) => {
+                      handleSelectedPerson(person.id);
+                    }}
+                    placeholder={messages.eventContactCard.selectPlaceholder()}
+                    selectedPerson={null}
+                    variant="outlined"
+                  />
+                </Box>
+              </Box>
+            )}
+          </Box>
+        }
+      >
+        {data.contact && (
+          <Button
+            onClick={() => {
+              showConfirmDialog({
+                onSubmit: () => {
+                  model.removeContact();
+                },
+                warningText: messages.eventContactCard.warningText(),
+              });
+            }}
+            size="small"
+            variant="text"
+          >
+            <CloseIcon />
```

Comment:
When you want an icon in a button, you can (and should) use the `startIcon` prop, like so:

```tsx
<Button startIcon={<Close/>}>Label here</Button>
```

☝️  I'm writing this from memory so I may have made a mistake somewhere, but I think the idea comes across.